test/suite/test_bug029.py

```diff
@@ -0,0 +1,110 @@
+
+#!/usr/bin/env python
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# [TEST_TAGS]
+# checkpoint:recovery
+# [END_TAGS]
+
+import wttest
+import os, shutil
+
+# test_bug029.py
+#
+# Test that WT correctly propogates the most recent checkpoint time
+# across restarts. We validate this by reproducing the original bug
+# from WT-9457: frequent checkpoints pushed the checkpoint clock time
+# into the future such that immediately after a restart a backup could
+# see its checkpoint deleted out from under it. The result was fatal
+# read errors when restoring the backup.
+
+class test_bug029(wttest.WiredTigerTestCase):
+    conn_config = ("cache_size=50MB")
+    uri = "table:test_bug029"
+    bigvalue = "WiredTiger" * 100
+    backup_dir = "backup_dir"
+
+    def add_data(self, uri, start, count):
+        cursor = self.session.open_cursor(uri, None)
+        for i in range(start, start + count): 
+            cursor[i] = self.bigvalue
+        cursor.close()
+
+    def test_bug029(self):
+        # Create and populate table
+        self.session.create(self.uri, "key_format=i,value_format=S")
+        self.add_data(self.uri, 0, 2000)
+        self.session.checkpoint()
+
+        # Force the checkpoint time forward with a lot of quick checkpoints.
+        for i in range(100):
+            self.session.checkpoint("force=1")
+
+        # Add more data and checkpoint again. The creates a bunch of pages
+        # in the final checkpoint that can be deleted and reused if we hit
+        # the bug.
+        self.add_data(self.uri, 2000, 2000)
+        self.session.checkpoint()
+
+        # Shutdown and reopen.
+        self.reopen_conn()
+
+        self.add_data(self.uri, 0, 100)
+
```

Comment:
I'd add a small comment about opening the backup cursor here and holding it in while we force more checkpoints. My eyes skimmed over it on my first read.

Reply:
Fixed in [04de73c](https://github.com/wiredtiger/wiredtiger/pull/8245/commits/04de73cd3345c280490cd5e336ac77226b5c37ed)